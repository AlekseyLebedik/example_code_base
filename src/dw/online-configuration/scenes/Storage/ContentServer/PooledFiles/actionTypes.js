export const STORAGE_POOLED_FILES_FETCH = 'storage/STORAGE_POOLED_FILES_FETCH';
export const STORAGE_POOLED_FILES_FETCH_SUCCESS =
  'storage/STORAGE_POOLED_FILES_FETCH_SUCCESS';

export const STORAGE_POOLED_FILES_LIST_ITEM_ONCLICK =
  'storage/STORAGE_POOLED_FILES_LIST_ITEM_ONCLICK';

export const STORAGE_POOLED_FILES_UPLOAD =
  'storage/STORAGE_POOLED_FILES_UPLOAD';
export const STORAGE_POOLED_FILES_UPLOAD_SUCCESS =
  'storage/STORAGE_POOLED_FILES_UPLOAD_SUCCESS';

export const STORAGE_POOLED_FILES_DELETE =
  'storage/STORAGE_POOLED_FILES_DELETE';
export const STORAGE_POOLED_FILES_DELETE_SUCCESS =
  'storage/STORAGE_POOLED_FILES_DELETE_SUCCESS';

export const STORAGE_POOLED_FILES_DOWNLOAD =
  'storage/STORAGE_POOLED_FILES_DOWNLOAD';
export const STORAGE_POOLED_FILES_DOWNLOAD_SUCCESS =
  'storage/STORAGE_POOLED_FILES_DOWNLOAD_SUCCESS';

export const STORAGE_POOLED_FILES_SUMMARY_DELETE =
  'storage/STORAGE_POOLED_FILES_SUMMARY_DELETE';
export const STORAGE_POOLED_FILES_SUMMARY_DELETE_SUCCESS =
  'storage/STORAGE_POOLED_FILES_SUMMARY_DELETE_SUCCESS';

export const STORAGE_POOLED_FILES_SUMMARY_DOWNLOAD =
  'storage/STORAGE_POOLED_FILES_SUMMARY_DOWNLOAD';
export const STORAGE_POOLED_FILES_SUMMARY_DOWNLOAD_SUCCESS =
  'storage/STORAGE_POOLED_FILES_SUMMARY_DOWNLOAD_SUCCESS';
