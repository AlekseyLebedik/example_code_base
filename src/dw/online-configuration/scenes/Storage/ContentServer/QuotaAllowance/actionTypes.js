export const STORAGE_QUOTA_ALLOWANCE_FETCH =
  'storage/STORAGE_QUOTA_ALLOWANCE_FETCH';
export const STORAGE_QUOTA_ALLOWANCE_FETCH_SUCCESS =
  'storage/STORAGE_QUOTA_ALLOWANCE_FETCH_SUCCESS';

export const STORAGE_QUOTA_ALLOWANCE_LIST_ITEM_ONCLICK =
  'storage/STORAGE_QUOTA_ALLOWANCE_LIST_ITEM_ONCLICK';

export const STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_OPEN =
  'storage/STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_OPEN';
export const STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_CLOSE =
  'storage/STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_CLOSE';
export const STORAGE_QUOTA_ALLOWANCE_ADD =
  'storage/STORAGE_QUOTA_ALLOWANCE_ADD';
export const STORAGE_QUOTA_ALLOWANCE_ADD_SUCCESS =
  'storage/STORAGE_QUOTA_ALLOWANCE_ADD_SUCCESS';
