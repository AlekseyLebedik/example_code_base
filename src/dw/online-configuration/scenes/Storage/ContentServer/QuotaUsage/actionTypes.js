export const STORAGE_QUOTA_USAGE_FETCH = 'storage/STORAGE_QUOTA_USAGE_FETCH';
export const STORAGE_QUOTA_USAGE_FETCH_SUCCESS =
  'storage/STORAGE_QUOTA_USAGE_FETCH_SUCCESS';

export const STORAGE_QUOTA_USAGE_LIST_ITEM_ONCLICK =
  'storage/STORAGE_QUOTA_USAGE_LIST_ITEM_ONCLICK';

export const STORAGE_QUOTA_USAGE_ADD_MODAL_OPEN =
  'storage/STORAGE_QUOTA_USAGE_ADD_MODAL_OPEN';
export const STORAGE_QUOTA_USAGE_ADD_MODAL_CLOSE =
  'storage/STORAGE_QUOTA_USAGE_ADD_MODAL_CLOSE';
export const STORAGE_QUOTA_USAGE_ADD = 'storage/STORAGE_QUOTA_USAGE_ADD';
export const STORAGE_QUOTA_USAGE_ADD_SUCCESS =
  'storage/STORAGE_QUOTA_USAGE_ADD_SUCCESS';
