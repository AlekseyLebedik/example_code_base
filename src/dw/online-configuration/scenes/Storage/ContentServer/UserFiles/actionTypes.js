export const STORAGE_USER_FILES_FETCH = 'storage/STORAGE_USER_FILES_FETCH';
export const STORAGE_USER_FILES_FETCH_SUCCESS =
  'storage/STORAGE_USER_FILES_FETCH_SUCCESS';

export const STORAGE_USER_FILES_LIST_ITEM_ONCLICK =
  'storage/STORAGE_USER_FILES_LIST_ITEM_ONCLICK';

export const STORAGE_USER_FILES_UPLOAD = 'storage/STORAGE_USER_FILES_UPLOAD';
export const STORAGE_USER_FILES_UPLOAD_SUCCESS =
  'storage/STORAGE_USER_FILES_UPLOAD_SUCCESS';

export const STORAGE_USER_FILES_DELETE = 'storage/STORAGE_USER_FILES_DELETE';
export const STORAGE_USER_FILES_DELETE_SUCCESS =
  'storage/STORAGE_USER_FILES_DELETE_SUCCESS';

export const STORAGE_USER_FILES_DOWNLOAD =
  'storage/STORAGE_USER_FILES_DOWNLOAD';
export const STORAGE_USER_FILES_DOWNLOAD_SUCCESS =
  'storage/STORAGE_USER_FILES_DOWNLOAD_SUCCESS';
