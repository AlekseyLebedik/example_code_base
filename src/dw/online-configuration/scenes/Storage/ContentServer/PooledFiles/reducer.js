import * as AT from './actionTypes';

const INITIAL_STATE = {
  entries: [],
  nextPageToken: undefined,
  selectedFile: undefined,
  elementsOrder: [],
  q: undefined,
  uploadedFiles: [],
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.STORAGE_POOLED_FILES_FETCH_SUCCESS: {
      let newEntries = [...action.entries];
      if (action.append && state.uploadedFiles.length !== 0) {
        state.uploadedFiles.forEach(fileID => {
          const idx = newEntries.findIndex(entry => fileID === entry.fileID);
          if (idx > -1) {
            newEntries = newEntries.splice(idx, 1);
          }
        });
      }
      return {
        ...state,
        entries: action.append ? [...state.entries, ...newEntries] : newEntries,
        nextPageToken: action.nextPageToken,
        elementsOrder: action.elementsOrder,
        q: action.q,
        uploadedFiles: action.append ? state.uploadedFiles : [],
      };
    }
    case AT.STORAGE_POOLED_FILES_LIST_ITEM_ONCLICK:
      return {
        ...state,
        selectedFile: action.file,
      };
    case AT.STORAGE_POOLED_FILES_UPLOAD_SUCCESS:
      return {
        ...state,
        entries: [...state.entries, action.file],
        uploadedFiles: [...state.uploadedFiles, action.file.fileID],
        selectedFile: action.file,
      };
    case AT.STORAGE_POOLED_FILES_DELETE_SUCCESS:
      return {
        ...state,
        entries: state.entries.filter(
          file => parseInt(file.fileID, 10) !== action.fileID
        ),
        uploadedFiles: state.uploadedFiles.filter(
          fileID => fileID !== action.fileID
        ),
        selectedFile: undefined,
      };
    case AT.STORAGE_POOLED_FILES_SUMMARY_DELETE_SUCCESS:
      return {
        ...state,
        entries: state.entries.map(entry => {
          if (entry.fileID !== action.fileID) {
            return entry;
          }
          return {
            ...entry,
            summaryFileSize: '0',
          };
        }),
        selectedFile: {
          ...state.selectedFile,
          summaryFileSize: '0',
        },
      };
    default:
      return state;
  }
};
