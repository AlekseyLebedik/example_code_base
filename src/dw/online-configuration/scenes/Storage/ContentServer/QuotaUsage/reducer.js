import * as AT from './actionTypes';

const INITIAL_STATE = {
  entries: [],
  nextPageToken: undefined,
  selectedListItem: undefined,
  elementsOrder: [],
  q: undefined,
  addModalVisible: false,
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.STORAGE_QUOTA_USAGE_FETCH_SUCCESS: {
      const existingIds = state.entries.map(item => item.userID);
      return {
        ...state,
        entries: action.append
          ? [
              ...state.entries,
              ...action.entries.filter(
                item => !existingIds.includes(item.userID)
              ),
            ]
          : action.entries,
        nextPageToken: action.nextPageToken,
        elementsOrder: action.elementsOrder,
        selectedListItem: action.append ? state.selectedListItem : undefined,
        q: action.q,
      };
    }
    case AT.STORAGE_QUOTA_USAGE_LIST_ITEM_ONCLICK:
      return {
        ...state,
        selectedListItem: action.listItem,
      };
    case AT.STORAGE_QUOTA_USAGE_ADD_MODAL_OPEN:
      return {
        ...state,
        addModalVisible: true,
      };
    case AT.STORAGE_QUOTA_USAGE_ADD_MODAL_CLOSE:
      return {
        ...state,
        addModalVisible: false,
      };
    case AT.STORAGE_QUOTA_USAGE_ADD_SUCCESS: {
      const idx = state.entries.findIndex(
        item => item.userID === action.listItem.userID
      );
      const newEntries = [...state.entries];
      if (idx > -1) {
        newEntries.splice(idx, 1, action.listItem);
      } else {
        newEntries.push(action.listItem);
      }
      return {
        ...state,
        entries: newEntries,
        addModalVisible: false,
      };
    }
    default:
      return state;
  }
};
