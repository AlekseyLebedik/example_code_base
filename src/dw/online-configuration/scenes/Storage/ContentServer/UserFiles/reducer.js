import * as AT from './actionTypes';

const INITIAL_STATE = {
  entries: [],
  nextPageToken: undefined,
  selectedFile: undefined,
  elementsOrder: [],
  q: undefined,
  uploadedFiles: [],
};

export const reducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.STORAGE_USER_FILES_FETCH_SUCCESS: {
      let newEntries = [...action.entries];
      if (action.append && state.uploadedFiles.length !== 0) {
        state.uploadedFiles.forEach(fileID => {
          const idx = newEntries.findIndex(entry => fileID === entry.fileID);
          if (idx > -1) {
            newEntries = newEntries.splice(idx, 1);
          }
        });
      }
      return {
        ...state,
        entries: action.append ? [...state.entries, ...newEntries] : newEntries,
        nextPageToken: action.nextPageToken,
        elementsOrder: action.elementsOrder,
        selectedFile: undefined,
        q: action.q,
        uploadedFiles: action.append ? state.uploadedFiles : [],
      };
    }
    case AT.STORAGE_USER_FILES_LIST_ITEM_ONCLICK:
      return {
        ...state,
        selectedFile: action.file,
      };
    case AT.STORAGE_USER_FILES_UPLOAD_SUCCESS:
      return {
        ...state,
        entries: [...state.entries, action.file],
        uploadedFiles: [...state.uploadedFiles, action.file.fileID],
        selectedFile: action.file,
      };
    case AT.STORAGE_USER_FILES_DELETE_SUCCESS:
      return {
        ...state,
        entries: state.entries.filter(file => file.fileID !== action.fileID),
        uploadedFiles: state.uploadedFiles.filter(
          fileID => fileID !== action.fileID
        ),
        selectedFile: null,
      };
    default:
      return state;
  }
};
