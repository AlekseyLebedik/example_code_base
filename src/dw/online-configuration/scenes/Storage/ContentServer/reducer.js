import { combineReducers } from 'redux';
import { reducer as pooledFilesReducer } from './PooledFiles/reducer';
import { reducer as userFilesReducer } from './UserFiles/reducer';
import { reducer as quotaAllowanceReducer } from './QuotaAllowance/reducer';
import { reducer as quotaUsageReducer } from './QuotaUsage/reducer';

export const reducer = combineReducers({
  PooledFiles: pooledFilesReducer,
  UserFiles: userFilesReducer,
  QuotaAllowance: quotaAllowanceReducer,
  QuotaUsage: quotaUsageReducer,
});
