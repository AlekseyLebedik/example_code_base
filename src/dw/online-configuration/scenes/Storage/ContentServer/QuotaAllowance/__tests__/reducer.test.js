import { reducer } from '../reducer';
import * as AT from '../actionTypes';

describe('QuotaAllowance', () => {
  describe('Reducer', () => {
    const initialState = reducer(undefined, {});
    const populatedState = {
      ...initialState,
      entries: [{ userID: 1 }, { userID: 2 }],
      selectedListItem: { userID: 1 },
      nextPageToken: 'ABC',
      elementsOrder: ['userID'],
    };

    it('returns default state', () => {
      const state = reducer(undefined, {});
      expect(state).toMatchSnapshot();
    });

    describe('STORAGE_QUOTA_ALLOWANCE_FETCH_SUCCESS', () => {
      const newEntries = [{ userID: 300 }, { userID: 400 }];

      it('replaces the entries with new ones when `append` is false, and update list metadata', () => {
        const action = {
          type: AT.STORAGE_QUOTA_ALLOWANCE_FETCH_SUCCESS,
          append: false,
          entries: newEntries,
          nextPageToken: 'YYZ',
          elementsOrder: ['userID'],
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });

      it('combines both entry lists when `append` is true, and update list metadata', () => {
        const action = {
          type: AT.STORAGE_QUOTA_ALLOWANCE_FETCH_SUCCESS,
          append: true,
          entries: newEntries,
          nextPageToken: 'YYZ',
          elementsOrder: ['userID'],
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('STORAGE_QUOTA_ALLOWANCE_LIST_ITEM_ONCLICK', () => {
      it('sets the selected item', () => {
        const action = {
          type: AT.STORAGE_QUOTA_ALLOWANCE_LIST_ITEM_ONCLICK,
          listItem: {
            userID: 2,
          },
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_OPEN', () => {
      it('sets modal visibility to `true`', () => {
        const action = {
          type: AT.STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_OPEN,
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_CLOSE', () => {
      it('sets modal visibility to `false`', () => {
        const closedState = { ...populatedState, addModalVisible: true };
        const action = {
          type: AT.STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_CLOSE,
        };
        const state = reducer(closedState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('STORAGE_QUOTA_ALLOWANCE_ADD_SUCCESS', () => {
      it('pushes new item to the entry list', () => {
        const action = {
          type: AT.STORAGE_QUOTA_ALLOWANCE_ADD_SUCCESS,
          listItem: {
            userID: 500,
          },
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });

      it('replaces existent item in the entry list', () => {
        const action = {
          type: AT.STORAGE_QUOTA_ALLOWANCE_ADD_SUCCESS,
          listItem: {
            userID: 2,
            newAttribute: 1,
          },
        };
        const state = reducer(populatedState, action);
        expect(state).toMatchSnapshot();
      });
    });
  });
});
