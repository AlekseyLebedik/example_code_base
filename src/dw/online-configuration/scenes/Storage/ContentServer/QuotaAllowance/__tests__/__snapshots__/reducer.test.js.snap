// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotaAllowance Reducer STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_CLOSE sets modal visibility to \`false\` 1`] = `
Object {
  "addModalVisible": false,
  "elementsOrder": Array [
    "userID",
  ],
  "entries": Array [
    Object {
      "userID": 1,
    },
    Object {
      "userID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "selectedListItem": Object {
    "userID": 1,
  },
}
`;

exports[`QuotaAllowance Reducer STORAGE_QUOTA_ALLOWANCE_ADD_MODAL_OPEN sets modal visibility to \`true\` 1`] = `
Object {
  "addModalVisible": true,
  "elementsOrder": Array [
    "userID",
  ],
  "entries": Array [
    Object {
      "userID": 1,
    },
    Object {
      "userID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "selectedListItem": Object {
    "userID": 1,
  },
}
`;

exports[`QuotaAllowance Reducer STORAGE_QUOTA_ALLOWANCE_ADD_SUCCESS pushes new item to the entry list 1`] = `
Object {
  "addModalVisible": false,
  "elementsOrder": Array [
    "userID",
  ],
  "entries": Array [
    Object {
      "userID": 1,
    },
    Object {
      "userID": 2,
    },
    Object {
      "userID": 500,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "selectedListItem": Object {
    "userID": 1,
  },
}
`;

exports[`QuotaAllowance Reducer STORAGE_QUOTA_ALLOWANCE_ADD_SUCCESS replaces existent item in the entry list 1`] = `
Object {
  "addModalVisible": false,
  "elementsOrder": Array [
    "userID",
  ],
  "entries": Array [
    Object {
      "userID": 1,
    },
    Object {
      "newAttribute": 1,
      "userID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "selectedListItem": Object {
    "userID": 1,
  },
}
`;

exports[`QuotaAllowance Reducer STORAGE_QUOTA_ALLOWANCE_FETCH_SUCCESS combines both entry lists when \`append\` is true, and update list metadata 1`] = `
Object {
  "addModalVisible": false,
  "elementsOrder": Array [
    "userID",
  ],
  "entries": Array [
    Object {
      "userID": 1,
    },
    Object {
      "userID": 2,
    },
    Object {
      "userID": 300,
    },
    Object {
      "userID": 400,
    },
  ],
  "nextPageToken": "YYZ",
  "q": undefined,
  "selectedListItem": Object {
    "userID": 1,
  },
}
`;

exports[`QuotaAllowance Reducer STORAGE_QUOTA_ALLOWANCE_FETCH_SUCCESS replaces the entries with new ones when \`append\` is false, and update list metadata 1`] = `
Object {
  "addModalVisible": false,
  "elementsOrder": Array [
    "userID",
  ],
  "entries": Array [
    Object {
      "userID": 300,
    },
    Object {
      "userID": 400,
    },
  ],
  "nextPageToken": "YYZ",
  "q": undefined,
  "selectedListItem": undefined,
}
`;

exports[`QuotaAllowance Reducer STORAGE_QUOTA_ALLOWANCE_LIST_ITEM_ONCLICK sets the selected item 1`] = `
Object {
  "addModalVisible": false,
  "elementsOrder": Array [
    "userID",
  ],
  "entries": Array [
    Object {
      "userID": 1,
    },
    Object {
      "userID": 2,
    },
  ],
  "nextPageToken": "ABC",
  "q": undefined,
  "selectedListItem": Object {
    "userID": 2,
  },
}
`;

exports[`QuotaAllowance Reducer returns default state 1`] = `
Object {
  "addModalVisible": false,
  "elementsOrder": Array [],
  "entries": Array [],
  "nextPageToken": undefined,
  "q": undefined,
  "selectedListItem": undefined,
}
`;
